### PR TITLE
Fix: serializers gather correctly and pass store

### PR DESCRIPTION
**Pull request by phlax**
This PR contains a cleanup and a couple of fixes

- Config.set_config does not return a Config object
- Config.none().get_config was not working correctly on ModelConfig
- Fix serializers should be called with Store, and gather on Project

and adjusts tests accordingly

### CODE REVIEW

**Comment by unho** [comment]
lgtm

File: tests/models/config.py [hunk 1]
```diff
@@ -37,8 +37,10 @@ def _test_config_bad_create(create_func):
 
 
 def _test_config_ob(conf, **kwargs):
-    conf.save()
```
Comment (ta2-1) [inline tests/models/config.py]: it overrides `conf` fuction parameter.

**Comment by ta2-1** [comment]
lgtm after comment is addressed.

File: tests/models/config.py (hunk 1 above)
Comment (phlax) [inline tests/models/config.py]: cool, ill remove

**Pull request merged**